### PR TITLE
crossdock: Run all behavior tests from test suite

### PR DESCRIPTION
**Pull request by abhinav**
This removes the need to reimplement the tests for each behavior.

### CODE REVIEW

**Comment by abhinav** [comment]
@breerly

**Comment by abhinav** [comment]
@yarpc/golang

File: crossdock/crossdock_test.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	type params map[string]string
+	type axes map[string][]string
+
+	defaultParams := params{"server": "127.0.0.1"}
+
+	behaviors := []struct {
```
Comment (breerly) [inline crossdock/crossdock_test.go]: wondering if we should consider parsing `docker-compose.yaml` for `AXIS_` and `BEHAVIOR_`...

**Comment by breerly** [comment]
This is a huge step in the right direction! 👍

File: crossdock/crossdock_test.go (hunk 1 above)
Comment (abhinav) [inline crossdock/crossdock_test.go]: I considered doing that but I dislike the idea of tying tests to that. I'm okay with doing this by-hand for now and switching to parsing if that becomes unscalable.

Comment (breerly) [inline crossdock/crossdock_test.go]: Agreed - this is a hell of alot better than how it was before. Let's use it this way for awhile and iterate 👍

**Pull request merged**